### DISCUSSION TITLE
v5.x Changes and Features

### DISCUSSION THREAD

**Discussion by Muhammad-Altabba** [ethereum/web3.js#6404] [General]
Pumping the version to 5.x will not be any time soon. However, it is good to have a list of the proposed changes and requested features. So, when the time comes for 5.x, we will already have a list of the "To-Do"s.

As a start, every code currently marked as `@deprecated ` should be no longer available in 5.x and the code around it may change. 

And, in this discussion, we put the links to an already created issue that detailed the feature or the change suggested, in order to keep track of them.

**Comment by Muhammad-Altabba** [comment]
Here is one needed chage:

- Update the returned value of `.request(...)` method at the providers to comply with EIP-1193. More info at:  https://github.com/web3/web3.js/issues/6345#issuecomment-1705985333

**Comment by mconnelly8** [comment]
Feature Request:

-  gasUsed values from getTransactionReceipt should be in wei to avoid confusion and rounding issues #6360 
-  Breaking Change: Support for polling #6092
